plugins/module_utils/k8s/runner.py

```diff
@@ -61,23 +63,30 @@ def run_module(module) -> None:
             if module.params.get("continue_on_error"):
                 result = {"error": "{0}".format(e)}
             else:
-                module.fail_json(msg=e)
+                module.fail_json(msg=to_native(e))
+
         if module.warnings:
             result["warnings"] = module.warnings
 
         results.append(result)
 
-    module.exit_json(**results)
+    module.exit_json(result=results)
```

Comment:
I believe this is the reason for the current failure. We need to replicate the behavior here: https://github.com/ansible-collections/kubernetes.core/blob/ed33d0b56ed61857b370a7a87d340578ab531862/plugins/module_utils/common.py#L778-L781